index.js

```diff
@@ -51,26 +54,23 @@ function RedisClient(stream, options) {
     this.should_buffer = false;
     this.command_queue_high_water = this.options.command_queue_high_water || 1000;
     this.command_queue_low_water = this.options.command_queue_low_water || 0;
-    this.max_attempts = null;
-    if (options.max_attempts && !isNaN(options.max_attempts) && options.max_attempts > 0) {
+    if (options.max_attempts && options.max_attempts > 0) {
         this.max_attempts = +options.max_attempts;
     }
     this.command_queue = new Queue(); // holds sent commands to de-pipeline them
     this.offline_queue = new Queue(); // holds commands issued but not able to be sent
     this.commands_sent = 0;
-    this.connect_timeout = false;
-    if (options.connect_timeout && !isNaN(options.connect_timeout) && options.connect_timeout > 0) {
+    if (options.connect_timeout && options.connect_timeout > 0) {
         this.connect_timeout = +options.connect_timeout;
     }
     this.enable_offline_queue = true;
-    if (typeof this.options.enable_offline_queue === "boolean") {
-        this.enable_offline_queue = this.options.enable_offline_queue;
+    if (this.options.enable_offline_queue === false) {
+        this.enable_offline_queue = false;
     }
     this.retry_max_delay = null;
-    if (options.retry_max_delay !== undefined && !isNaN(options.retry_max_delay) && options.retry_max_delay > 0) {
-        this.retry_max_delay = options.retry_max_delay;
+    if (options.retry_max_delay && options.retry_max_delay > 0) {
+        this.retry_max_delay = +options.retry_max_delay;
```

Comment:
The + is only a type coercion to number :)
`a > 0` is also doing a type coercion, so it'll be the same as it was before